pyat/at/tracking/particles.py

```diff
@@ -116,46 +167,43 @@ def sigma_matrix(ring=None, twiss_in=None, emitx=None,
     longitudinal parameters, then the 2x2 uncorrelated
     matrices are computed for each plane and combined
     into the 6x6 matrix.
-    """
-    flag = emitx or emity or blength or espread
-    if flag:
-        assert emitx is not None, 'emitx must be defined'
-        assert emity is not None, 'emity must be defined'
-        assert blength is not None, 'blength must be defined'
-        assert espread is not None, 'espread must be defined'
 
-    if ring:
-        ld0, bd, ld = ring.get_optics()
-        if ring.radiation and not flag:
-            print('Generating correlated sigma matrix using ohmi envelope')
-            emit0, beamdata, emit = ohmi_envelope(ring, refpts=[0])
-            sig_matrix = emit.r66[0]
-        elif flag:
-            print('Generating pseudo-correlated matrix '
-                  'from initial optics conditions')
-            if ring.radiation:
-                print('Ignoring provided blength and '
-                      'calculating it based on espread')
-                blength = _compute_bunch_length_from_espread(ring, espread)
-            sig_matrix = _generate_sigma_matrix(ld0, emitx, emity,
-                                                blength, espread,
-                                                ring.radiation)
-        else:
-            raise AttributeError('Radiation is off but no '
-                                 'emittances are specified')
-    elif twiss_in:
-        if not emitx:
-            raise AttributeError('Emittances must be '
-                                 'specified for twiss_in')
-        print('Generating un-correlated sigma matrix '
-              'from parameters in twiss_in')
-        sig_matrix = _generate_sigma_matrix(twiss_in, emitx,
-                                            emity, blength,
-                                            espread, False)
+    If neither a lattice object nor a twiss_in is provided,
+    then the beta, alpha and emittance for horizontal and 
+    vertical is required, as well as blength and espread.
+    This then computes the analytical uncoupled sigma matrix
+    """
 
+    d = kwargs
+    betax = d.get('betax', None)
+    alphax = d.get('betax', None)
+    emitx = d.get('emitx', None)
+    betay = d.get('betay', None)
+    alphay = d.get('betay', None)
+    emity = d.get('emity', None)
+    blength = d.get('blength', None)
+    espread = d.get('espread', None)
+    verbose = d.get('verbose', False)
+
+    if isinstance(ring, list) or isinstance(twiss_in, numpy.recarray):
```

Comment:
Why not if ring is not None or twiss_in is not None? If they have the wrong type containing functions will raise exceptions

Reply:
done